### DISCUSSION TITLE
Remove, or expire metric from being exported? Or dropping specific attributes?

### DISCUSSION THREAD

**Discussion by EStork09** [open-telemetry/opentelemetry-go#4619] [Q&A] [answered]
Hello, I have some metrics that I plan to add, and it might be an antipattern, but ideally I only want to report the data for a certain amount of time then drop it when it isn't useful, rather than doing a 1 or 0. Is there a way with a view or something that you can discard a metric after a period of time? I think I have seen similar questions but I am unsure if they were going after the exact same thing. But ideally it would free up the memory rather than constantly firing all the time when the data source no longer exists. I could arguably restart the entire application to free it up, but that sounds dumb.

**Comment by pellared** [comment]
My quick idea it to shutdown old meter provider and create a new one.

**Reply by EStork09** [reply]
Yeah that is fair, it seems rather odd, every 15 minutes or so recycle the meter provider. Would that be a bad practice though?

**Reply by EStork09** [reply]
I thought about looking into this more:

```
view := metric.NewView(
  metric.Instrument{
    Name:  "latency",
    Scope: instrumentation.Scope{Name: "http"},
  },
  metric.Stream{Aggregation: metric.AggregationDrop{}},
)
```

But I am doubting the the aggregationdrop would do what I am thinking, I assume it would drop everything but the newest one.

**Comment by MrAlias** [comment, marked as answer]
This is an OpenTelemetry wide [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/3062), for what it's worth. In that issue @dashpole [made a proposal](https://github.com/open-telemetry/opentelemetry-specification/issues/3062#issuecomment-1412306635) on how to resolve this using asynchronous instruments that I think you could try here.

**Reply by EStork09** [reply]
I think I see the idea there, that worked for me, thanks for the heads up about the overarching issue, but that asynchronous method worked pretty well!
